tests/backup/backup_orphaned_delete_test.go

```diff
@@ -0,0 +1,505 @@
+					break
+				}
+				mu.Lock()
+				defer mu.Unlock()
+				userSuccessfulBackupMap[backupName] = namespace
+			}
+			for _, namespace := range appNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					backupName := fmt.Sprintf("%s-%s-%d-%v", BackupNamePrefix, namespace, i, time.Now().Unix())
+					wg.Add(1)
+					go createBackup(backupName, namespace)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of successful user backups taken are: %v", userSuccessfulBackupMap)
+		})
+		Step(fmt.Sprintf("Verify successful backups of the user %s from the admin", infraAdminUser), func() {
+			log.InfoD(fmt.Sprintf("Verifying successful backups of the user %s from the admin", infraAdminUser))
+			nonAdminCtx, err := backup.GetNonAdminCtx(infraAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", infraAdminUser)
+			userOwnerID, err := portworx.GetSubFromCtx(nonAdminCtx)
+			log.FailOnError(err, "failed to fetch user owner id %s", infraAdminUser)
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			backupNamesByOwnerID, err := GetAllBackupNamesByOwnerID(userOwnerID, orgID, ctx)
+			log.FailOnError(err, "failed to fetch backup names with owner id %s from the admin", userOwnerID)
+			for backupName := range userSuccessfulBackupMap {
+				if !IsPresent(backupNamesByOwnerID, backupName) {
+					err := fmt.Errorf("successful backup %s is not listed in backup names %s", backupName, backupNamesByOwnerID)
+					log.FailOnError(fmt.Errorf(""), err.Error())
+				}
+			}
+		})
+		Step(fmt.Sprintf("Take in progress restore of backups from the user %s", infraAdminUser), func() {
+			log.InfoD(fmt.Sprintf("Taking in progress restore of backups from the user %s", infraAdminUser))
+			nonAdminCtx, err := backup.GetNonAdminCtx(infraAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", infraAdminUser)
+			var wg sync.WaitGroup
+			var mu sync.RWMutex
+			createRestore := func(backupName string, restoreName string, namespace string) {
+				defer GinkgoRecover()
+				defer wg.Done()
+				namespaceMapping := map[string]string{namespace: fmt.Sprintf("r-%s-%v", namespace, time.Now().Unix())}
+				_, err = CreateRestoreWithoutCheck(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, nonAdminCtx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of restore %s of backup %s", restoreName, backupName))
+				mu.Lock()
+				defer mu.Unlock()
+				userInProgressRestoreNames = append(userInProgressRestoreNames, restoreName)
+			}
+			for backupName, namespace := range userSuccessfulBackupMap {
+				wg.Add(1)
+				restoreName := fmt.Sprintf("%s-%s", restoreNamePrefix, backupName)
+				go createRestore(backupName, restoreName, namespace)
+			}
+			wg.Wait()
+			log.Infof("The list of in progress user restores taken are: %v", userInProgressRestoreNames)
```

Comment:
How are we making sure that the restore is in progress ?

Reply:
The CreateRestoreWithoutCheck function will not wait for the restores to be successful. I am ensuring that the restores are in in-progress while deleting them.